experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,265 @@
+# I would like to credit Dario Mathiä who produced an initial version of the following code
+
+import Chevie: complex_reflection_group, charinfo, CharTable, CyclotomicNumbers
+export wreath_macs
+
+# Tools
+
+#gives indices of a subarray
+function subarray_indices(sub,arr)
+  res = Int[]
+  for i in 1:length(sub)
+    push!(res,findfirst(x->x==sub[i],arr))
+  end
+  return res
+end
+
+# Computes the b-invariant of a partition
+function b1(lambda::Partition)
+  if length(lambda)==0
+    return 0
+  end
+  return sum(i->(i-1)*lambda[i],1:length(lambda))
+end
+
+# Computes the b-invariant of a multipartition
+function btot(lbb::Multipartition)
+  return b1(map(lambda -> sum(lambda)[1], lbb)) + length(lbb)*sum(i -> b1(lbb[i]),1:length(lbb))
+end
+
+function beta_number_to_partition(beta::Vector{Integer})
+  lb=Integer[]
+  lbeta=length(beta)
+  for j in 1:lbeta
+    nbholes=beta[j]-beta[1]-(j-1)
+    if nbholes >= 1
+    pushfirst!(lb, nbholes)
+    end
+  end
+  return partition(lb)
+end
+
+function inv_perm(wperm::PermGroupElem, r::Integer)
```

Comment:
Doesn't `inv(wperm)` work for you here? If not, it would be great to hear why. And otherwise this function could be removed with that replacement